### PR TITLE
NO-JIRA: address malformed CMO configmap post-test

### PR DESCRIPTION
**Pull request by rexagod**

### CODE REVIEW

**Comment by rexagod** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-metal-ipi-ovn-dualstack-techpreview

File: test/extended/prometheus/collection_profiles.go [hunk 1]
```diff
@@ -84,14 +84,12 @@ var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfil
 			if err != nil {
 				if errors.IsNotFound(err) {
 					g.By("initially, creating a configuration for the operator as it did not exist")
+					operatorConfiguration = nil
 					err = r.makeCollectionProfileConfigurationFor(tctx, collectionProfileDefault)
```
Comment (simonpasquier) [inline test/extended/prometheus/collection_profiles.go:88]: With the 2 other suggestions, I find the code flow to be easier to read.
```suggestion
					return r.makeCollectionProfileConfigurationFor(tctx, collectionProfileDefault)
```

File: test/extended/prometheus/collection_profiles.go [hunk 2]
```diff
@@ -84,14 +84,12 @@ var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfil
...
 				}
```
Comment (simonpasquier) [inline test/extended/prometheus/collection_profiles.go:89]:
```suggestion
				}
				return err
```

File: test/extended/prometheus/collection_profiles.go [hunk 3]
```diff
@@ -84,14 +84,12 @@ var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfil
...
-				if err != nil {
-					return err
-				}
 			}
 
-			return nil
+			return err
```
Comment (simonpasquier) [inline test/extended/prometheus/collection_profiles.go:92]:
```suggestion
			return nil
```

**Comment by rexagod** [comment]
Thank you for the review, Simon. I'm keeping an eye out for the `payload-job` above and will post the results as soon as that's done. 🤞🏼 for a green CI.

File: test/extended/prometheus/collection_profiles.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ type runner struct {
 var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfiles] The collection profiles feature-set", g.Ordered, func() {
 	defer g.GinkgoRecover()
 
-	o.SetDefaultEventuallyTimeout(15 * time.Minute)
+	o.SetDefaultEventuallyTimeout(20 * time.Minute)
```
[the remark this answers is not in the document]
Comment (rexagod) [reply test/extended/prometheus/collection_profiles.go:61]: (for instance, we've seen ~5 minutes for this [here](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-origin-29297-nightly-4.18-e2e-metal-ipi-ovn-ipv6-techpreview/1857206525055995904) and ~7 minutes [here](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-origin-29297-nightly-4.18-e2e-metal-ipi-ovn-dualstack-techpreview/1857206520714891264))

**Comment by rexagod** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-aws-ovn-techpreview-serial

**Comment by rexagod** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-metal-ipi-ovn-dualstack-techpreview

**Comment by rexagod** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-metal-ipi-ovn-ipv6-techpreview

**Comment by rexagod** [comment]
Test results on [metal-dualstack-techpreview](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-origin-29309-nightly-4.18-e2e-metal-ipi-ovn-dualstack-techpreview/1859512943918452736) and [metal-ipv6-techpreview](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-origin-29309-nightly-4.18-e2e-metal-ipi-ovn-ipv6-techpreview/1859513091201437696) seem okay for the ones the concern us.

**Comment by xueqzhan** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.19-e2e-aws-ovn-techpreview

**Comment by rexagod** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.19-e2e-gcp-ovn-crun-techpreview

**Pull request merged**